src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/General/capacityUtils.js

```diff
@@ -0,0 +1,289 @@
+  htmlType: (modificationType) =>
+    modificationType !== list && INPUT_TYPES.HIDDEN,
+  validation: lazy((_, { context }) =>
+    array(number().positive())
+      .when(`$general.${type}`, {
+        is: (modificationType) => modificationType === list,
+        then: (schema) => schema.required().min(1),
+        otherwise: (schema) => schema.notRequired().nullable(),
+      })
+      .default(() => {
+        const capacityUserInput = context.extra?.USER_INPUTS?.[fieldName]
+        const { options = [] } = getUserInputParams(capacityUserInput)
+        const numberOpts = options?.filter((opt) => opt !== ' ' && !isNaN(+opt))
+
+        return numberOpts
+      })
+  ),
+  fieldProps: { freeSolo: true },
+  grid: { sm: 12, md: 12 },
+})
+
+/**
+ * @param {CapacityFieldName} fieldName - Capacity field name
+ * @returns {Field[]} - Fields for capacity modification
+ */
+export const generateModificationInputs = (fieldName) => {
+  const modificationIds = getModificationIdsByFieldName(fieldName)
+
+  return [
+    modificationTypeInput(fieldName, modificationIds),
+    modificationOptionsInput(fieldName, modificationIds),
+    ...modificationRangeInputs(fieldName, modificationIds),
+  ]
+}
+
+/**
+ * @param {Field} hrField - Hot resize field
+ * @param {CapacityFieldName} hrField.fieldName - Capacity field name
+ * @param {Field} maxField - Max field
+ * @returns {Field[]} Hot resize fields
+ */
+export const generateHotResizeInputs = (
+  { name: hrFieldName, ...hrField },
+  maxField
+) => [
+  {
+    ...hrField,
+    name: `HOT_RESIZE.${hrFieldName}`,
+    dependOf: `HOT_RESIZE.${hrFieldName}`,
+    label: T.EnableHotResize,
+    type: INPUT_TYPES.SWITCH,
+    validation: boolean().yesOrNo(),
+    grid: { sm: 12, md: 5 },
+  },
+  {
+    ...maxField,
+    dependOf: `HOT_RESIZE.${hrFieldName}`,
+    type: INPUT_TYPES.TEXT,
+    htmlType: (enabledHr) => (enabledHr ? 'number' : INPUT_TYPES.HIDDEN),
+    validation: number()
+      .positive()
+      .default(() => undefined)
+      .when(`HOT_RESIZE.${hrFieldName}`, (enabledHr, schema) =>
+        enabledHr ? schema.required() : schema.notRequired()
+      ),
+    grid: { sm: 12, md: 7 },
+  },
+]
+
+/**
+ * @param {Field} config - Configuration
+ * @param {CapacityFieldName} config.name - Capacity field name
+ * @param {BaseSchema} config.validation - Validation schema
+ * @returns {Field} - Field with validation modification conditions
+ */
+export const generateCapacityInput = ({ validation, ...field }) => ({
+  ...field,
+  type: INPUT_TYPES.TEXT,
+  htmlType: 'number',
+  validation: validation
+    .when(`$general.MODIFICATION.${field.name}.type`, {
+      is: (modificationType) =>
+        modificationType === range || modificationType === rangeFloat,
+      then: (schema) =>
+        schema
+          .max(ref(`$general.MODIFICATION.${field.name}.max`))
+          .min(ref(`$general.MODIFICATION.${field.name}.min`)),
+      otherwise: (schema) => schema,
+    })
+    .when(
+      [
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).